dask/array/core.py

```diff
@@ -1573,9 +1573,14 @@ def __float__(self):
     def __complex__(self):
         return self._scalarfunc(complex)
 
+    def __index__(self):
+        return self._scalarfunc(int)
+
     def __setitem__(self, key, value):
-        from .routines import where
+        from .routines import where  # , count_nonzero
+        from .slicing import parse_assignment_indices, setitem
```

Comment:
The where function is not used in your parts right? Then that's an unnecessary import when `isinstance(key, Array) = False`. Perhaps move that row down to just before the `try: y = where(key, value, self)`?

Same idea with parse_assignment and setitem. Those are not needed when `isinstance(key, Array) = True` Although since .slicing has already been loaded at the top of the module so maybe parse_assignment_indices, setitem should just be defined at the top as well?